compiler/src/dmd/mars.d

```diff
@@ -308,17 +308,21 @@ const(char)[] parse_arch_arg(Strings* args, const(char)[] arch)
 {
     foreach (const p; *args)
     {
-        const(char)[] arg = p.toDString;
+        const arg = p.toDString;
 
-        if (arg.length && arg[0] == '-')
+        switch (arg)
         {
-            if (arg[1 .. $] == "m32" || arg[1 .. $] == "m64")
-                arch = arg[2 .. $];
-            else if (arg[1 .. $] == "m32mscoff")
-                arch = "32";
-            else if (arg[1 .. $] == "run")
+            case "-m32":
+            case "-m64":
+            case "-m32mscoff":
+                arch = arg[2 .. 4];
+                continue;
+            case "run":   // end of args to dmd
```

Comment:
`case "run"` should probably be `case "-run"` to be equivalent to the old code?

Reply:
Ooh, good catch

Reply:
yes, you're right. But I borked up my copy of the branch, and don't know how to revert it to what's here on github.

Reply:
ok I found a clumsy way to get it back on track